src/schemathesis/cli/junitxml.py

```diff
@@ -0,0 +1,42 @@
+import platform
+from typing import List, Optional
+
+import attr
+from click.utils import LazyFile
+from junit_xml import TestCase, TestSuite, to_xml_report_file
+
+from ..models import Status
+from ..runner import events
+from .handlers import EventHandler, ExecutionContext
+from .output import default
+
+
+@attr.s(slots=True)
+class JunitXMLHandler(EventHandler):
+    file_handle: LazyFile = attr.ib()
+    test_cases: List = attr.ib(factory=list)
+    start_time: Optional[float] = attr.ib(default=None)
+
+    def handle_event(self, context: ExecutionContext, event: events.ExecutionEvent) -> None:
+        if isinstance(event, events.Initialized):
+            self.start_time = event.start_time
+        if isinstance(event, events.BeforeExecution):
+            self.test_cases.append(TestCase(f"{event.method} {event.path}"))
```

Comment:
I think it could be done in `AfterExecution` branch below since this data is available there. To work on the same item without using `test_cases[-1]`.